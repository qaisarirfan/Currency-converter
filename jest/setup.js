jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useRoute: () => jest.fn(),
    useNavigation: () => ({
      navigation: () => jest.fn(),
      canGoBack: () => jest.fn(),
      goBack: () => jest.fn(),
      push: () => jest.fn(),
    }),
  };
});

jest.mock('react-native-bootsplash', () => ({
  hide: jest.fn(),
  show: jest.fn(),
}));

jest.mock('react-native-localize', () => ({
  getLocales: () => [
    { countryCode: 'GB', languageTag: 'en-GB', languageCode: 'en', isRTL: false },
    { countryCode: 'US', languageTag: 'en-US', languageCode: 'en', isRTL: false },
    { countryCode: 'FR', languageTag: 'fr-FR', languageCode: 'fr', isRTL: false },
  ],

  getNumberFormatSettings: () => ({
    decimalSeparator: '.',
    groupingSeparator: ',',
  }),

  getCalendar: () => 'gregorian', // or "japanese", "buddhist"
  getCountry: () => 'US', // the country code you want
  getCurrencies: () => ['USD', 'EUR'], // can be empty array
  getTemperatureUnit: () => 'celsius', // or "fahrenheit"
  getTimeZone: () => 'Europe/Paris', // the timezone you want
  uses24HourClock: () => true,
  usesMetricSystem: () => true,

  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}));

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = require('react-native').ScrollView;
  return { KeyboardAwareScrollView };
});

jest.mock('react-native-gesture-handler', () => {
  // eslint-disable-next-line global-require
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
