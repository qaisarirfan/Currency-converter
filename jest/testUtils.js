import React from "react"
import {render} from "react-native-testing-library"
import {Provider} from "react-redux"
import configureStore from "redux-mock-store"
import ThemeContextProvider from "../src/ContextUtils/ThemeContext"
import ConversionContextProvider from "../src/ContextUtils/ConversionContext"
import {initialState as themesState} from "../src/redux/reducers/themes"

const INITIAL_STATE = {
  themes: themesState,
}
const mockStore = configureStore([])
const store = mockStore(INITIAL_STATE)

const reduxRender = (ui, options) =>
  render(ui, {
    wrapper: ({children}) => (
      <Provider store={store}>
        <ThemeContextProvider>
          <ConversionContextProvider>{children}</ConversionContextProvider>
        </ThemeContextProvider>
      </Provider>
    ),
    ...options,
  })

// re-export everything for convenience
export * from "react-native-testing-library"

// override render method
export {reduxRender as render, store}
