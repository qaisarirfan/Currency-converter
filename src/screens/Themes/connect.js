import {connect} from "react-redux"

import {selectThemes} from "../../redux/reducers/themes/selectors"

const mapStateToProps = (state) => ({
  themes: selectThemes(state),
})

const mapDispatchToProps = {}

export default connect(mapStateToProps, mapDispatchToProps)
