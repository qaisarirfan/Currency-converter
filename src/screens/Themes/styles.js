import { StyleSheet } from 'react-native';

// Themes: Styling here
const styles = (styleableTheme) =>
  StyleSheet.create({
    root: {
      backgroundColor: styleableTheme[500],
      flex: 1,
    },
  });

export default styles;
