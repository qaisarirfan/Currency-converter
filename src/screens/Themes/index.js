import React, {useContext} from "react"
import {View, SafeAreaView, ScrollView} from "react-native"
import PropTypes from "prop-types"
import flow from "lodash/flow"
import Entypo from "react-native-vector-icons/Entypo"
import {useTranslation} from "react-i18next"

import themeStyles from "./styles"
import connect from "./connect"
import {RowItem, RowSeparator} from "../../components/RowItem"
import {HeaderBar} from "../../components/HeaderBar"
import {ThemeContext} from "../../ContextUtils/ThemeContext"

// Themes Component content
export const Themes = ({themes}) => {
  const {styleableTheme, changeTheme, defaultTheme} = useContext(ThemeContext)
  const styles = themeStyles(styleableTheme)
  const {t} = useTranslation()
  const colors = Object.keys(themes) || []

  return (
    <View style={styles.root}>
      <HeaderBar title={t("themes.title")} />
      <SafeAreaView>
        <ScrollView>
          {colors.map((color, index) => (
            <React.Fragment key={`theme-${index + 1}`}>
              <RowItem
                testID={`theme-${index + 1}`}
                title={`${t("themes.theme")} ${index + 1}`}
                onPress={() => changeTheme(color)}
                rightIcon={
                  <View
                    style={{
                      flexDirection: "row",
                    }}>
                    {defaultTheme === color ? (
                      <View style={{marginRight: 12}}>
                        <Entypo
                          name="check"
                          size={20}
                          color={styleableTheme[50]}
                        />
                      </View>
                    ) : null}
                    <View
                      style={{
                        width: 20,
                        height: 20,
                        backgroundColor: color,
                      }}
                    />
                  </View>
                }
              />
              <RowSeparator />
            </React.Fragment>
          ))}
        </ScrollView>
      </SafeAreaView>
    </View>
  )
}

// Themes Proptypes
Themes.propTypes = {
  themes: PropTypes.object,
}

// Themes Default props
Themes.defaultProps = {
  themes: {},
}

export default flow([connect])(Themes)
