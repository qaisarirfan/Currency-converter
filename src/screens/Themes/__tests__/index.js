import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import Themes from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<Themes />", () => {
  const createProps = () => ({})
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<Themes {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
