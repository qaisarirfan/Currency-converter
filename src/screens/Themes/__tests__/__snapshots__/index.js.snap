// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Themes /> Does not explode 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#4f6d7a",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "#36505d",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 20,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexGrow": 1,
            "justifyContent": "space-between",
            "paddingBottom": 12,
            "paddingTop": 12,
          },
          Object {
            "backgroundColor": "#36505d",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "justifyContent": "center",
            "paddingLeft": 6,
            "width": 250,
          }
        }
      >
        <RNGestureHandlerButton
          collapsable={false}
          onGestureEvent={[Function]}
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          onHandlerStateChange={[Function]}
          rippleColor={0}
          testID="back"
        >
          <View
            accessible={true}
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#eaedef",
                    "fontSize": 16,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Entypo",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï…‚
            </Text>
            <Text
              style={
                Object {
                  "color": "#eaedef",
                  "fontSize": 18,
                  "lineHeight": 18,
                }
              }
            >
              common.back
            </Text>
          </View>
        </RNGestureHandlerButton>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#36505d",
            "justifyContent": "center",
            "width": 250,
          }
        }
      >
        <Text
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#eaedef",
              "fontSize": 18,
              "lineHeight": 18,
            }
          }
        >
          themes.title
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "justifyContent": "center",
            "paddingRight": 6,
            "width": 250,
          }
        }
      />
    </View>
  </View>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <RCTScrollView>
      <View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="theme-1"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            themes.theme 1
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#2fbe74",
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="theme-2"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            themes.theme 2
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#8b50da",
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="theme-3"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            themes.theme 3
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#f69400",
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="theme-4"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            themes.theme 4
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#000000",
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#8499a2",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 16,
            }
          }
          testID="theme-5"
        >
          <Text
            style={
              Object {
                "color": "#eaedef",
                "fontSize": 16,
              }
            }
          >
            themes.theme 5
          </Text>
          <View
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "marginRight": 12,
                }
              }
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#eaedef",
                      "fontSize": 20,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Entypo",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                ï…€
              </Text>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#4f6d7a",
                  "height": 20,
                  "width": 20,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 0.5,
            }
          }
        />
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
